src/Sulu/Bundle/ContentBundle/Markup/Link/PageLinkProvider.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ContentBundle\Markup\Link;
+
+use Sulu\Component\Content\Repository\Content;
+use Sulu\Component\Content\Repository\ContentRepositoryInterface;
+use Sulu\Component\Content\Repository\Mapping\MappingBuilder;
+use Sulu\Component\Webspace\Manager\WebspaceManagerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Integrates pages into link-system.
+ */
+class PageLinkProvider implements LinkProviderInterface
+{
+    /**
+     * @var ContentRepositoryInterface
+     */
+    protected $contentRepository;
+
+    /**
+     * @var WebspaceManagerInterface
+     */
+    protected $webspaceManager;
+
+    /**
+     * @var RequestStack
+     */
+    protected $requestStack;
+
+    /**
+     * @var string
+     */
+    protected $environment;
+
+    /**
+     * @param ContentRepositoryInterface $contentRepository
+     * @param WebspaceManagerInterface $webspaceManager
+     * @param RequestStack $requestStack
+     * @param string $environment
+     */
+    public function __construct(
+        ContentRepositoryInterface $contentRepository,
+        WebspaceManagerInterface $webspaceManager,
+        RequestStack $requestStack,
+        $environment
+    ) {
+        $this->contentRepository = $contentRepository;
+        $this->webspaceManager = $webspaceManager;
+        $this->requestStack = $requestStack;
+        $this->environment = $environment;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfiguration()
+    {
+        return new LinkConfiguration('content.ckeditor.page-link', 'ckeditor/link/page@sulucontent');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function preload(array $hrefs, $locale, $published = true)
```

Comment:
yes it will collect all links on a page and divide it per provider and call it once.